tools/mpremote/mpremote/commands.py

```diff
@@ -300,6 +300,15 @@ def _mkdir(a, *b):
         do_filesystem_cp(state, src_path_joined, dest_path_joined, False, check_hash)
 
 
+def do_filesystem_recursive_rm(state, path):
+    if state.transport.fs_isdir(path):
+        for entry in state.transport.fs_listdir(path):
+            do_filesystem_recursive_rm(state, _remote_path_join(path, entry.name))
+        state.transport.fs_rmdir(path)
```

Comment:
The argument is a directory, and this function removes that directory; I'd argue that this should therefore be part of the `rmdir` command rather than the `rm` command.